### DISCUSSION TITLE
Shared knex connection pool

### DISCUSSION THREAD

**Discussion by Tornquist** [prisma/prisma#9615] [Q&A] [answered]
There are a number of actions that I need to perform that I'm unable to do with prisma directly. I have a lot of circular dependencies that make the various transactional creation non-viable.

I'd like to use knex for these, as I can get intermediate IDs on inserts/updates and perform all of the required actions in one transaction. Is there any way to get access to the inner prisma connection pooling so that I can use one connection pool for both tools?

The alternative is to write all of these critical actions as postgres functions, and just use prisma to execute them, but I'd like to avoid that for now if I can.

**Comment by janpio** [comment, marked as answer]
No, the connection pool is not and can not be accessible unfortunately, as that is owned by the Prisma Query Engine, which is implemented in Rust. Sorry. You can use the same connection pool via raw queries in Prisma Client of course, but that will probably not help in your situation.

Can you make sure there is an issue for your use case though with a proper description what you are trying to achieve and what is not working? Anything you need to fall back to Knex, is something we potentially want to avoid and improve.

**Comment by Tornquist** [comment]
My mistake, I had not seen the in-preview interactive transactions. I can do everything within prisma.

**Reply by janpio** [reply]
Ohhhh, sorry I missed that in your message. Let us know if this indeed fixes all your problems, that would be amazing 👍

**Reply by Tornquist** [reply]
So far it has resolved my problems! I've ported a section of my application and it's working for all of the dependent/related transactions, and for the raw queries that did not require a transaction. 

All looks good! 

I'll open new issues/tickets should there be any specific limitations that I find.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
